### PR TITLE
Add analytics events for giving portal (plus a couple of other small bug fixes)

### PR DESCRIPTION
**Pull request by Will-Howard**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205863372821550) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/ea-forum/giving-portal/DonationOpportunity.tsx [hunk 1]
```diff
@@ -87,25 +88,39 @@ const DonationOpportunity = ({candidate, classes}: {
 }) => {
   const {name, logoSrc, description} = candidate;
   return (
-    <div className={classes.root}>
-      <Link to={candidate.href}>
-        <div className={classes.header}>
-          <div className={classes.imageContainer}>
-            <img src={logoSrc} className={classes.image} />
+    <AnalyticsContext pageElementContext="donationOpportunity">
```
Comment (Will-Howard) [inline packages/lesswrong/components/ea-forum/giving-portal/DonationOpportunity.tsx:91]: This is mainly adding this context, you might want to look at it side by side

File: packages/lesswrong/components/ea-forum/giving-portal/ElectionFundCTA.tsx [hunk 2]
```diff
@@ -62,6 +63,7 @@ const styles = (theme: ThemeType) => ({
     borderRadius: theme.borderRadius.small,
     padding: 16,
     outline: "none",
+    whiteSpace: "nowrap",
```
Comment (Will-Howard) [inline packages/lesswrong/components/ea-forum/giving-portal/ElectionFundCTA.tsx:66]: Separate fix for the text wrapping. This does make the rightmost one a bit wider than the others, but only on a very narrow range of screen sizes so I think this is better

File: packages/lesswrong/components/ea-forum/giving-portal/EAGivingPortalPage.tsx [hunk 3]
```diff
@@ -302,19 +302,22 @@ const EAGivingPortalPage = ({classes}: {classes: ClassesType}) => {
...
   const toggleNotifyWhenVotingOpens = useCallback(() => {
-    // TODO: captureEvent
+    captureEvent('toggleNotifyWhenVotingOpens', { notifyForVotingOn: !notifyForVotingOn })
+
     if (!currentUser) {
       openDialog({
         componentName: "LoginPopup",
         componentProps: {},
       })
+      return;
```
Comment (Will-Howard) [inline packages/lesswrong/components/ea-forum/giving-portal/EAGivingPortalPage.tsx:315]: Unrelated: fix for this toggle changing when logged out (https://app.asana.com/0/1205806305763861/1205853507203268/f)

File: packages/lesswrong/lib/eaGivingSeason.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import moment from "moment";
 
 export const eaGivingSeason23ElectionName = "givingSeason23";
 
-export const donationElectionLink = "https://forum.effectivealtruism.org/posts/hAzhyikPnLnMXweXG/participate-in-the-donation-election-and-the-first-weekly";
+export const donationElectionLink = "/posts/hAzhyikPnLnMXweXG/participate-in-the-donation-election-and-the-first-weekly";
```
Comment (Will-Howard) [inline packages/lesswrong/lib/eaGivingSeason.ts:5]: Unrelated: relative link seems more consistent with other internal links

**Review by jpaddison3 — APPROVED** [review]

File: packages/lesswrong/components/ea-forum/giving-portal/Timeline.tsx [hunk 5]
```diff
@@ -184,7 +184,7 @@ const Timeline = ({
   return (
     <div className={classNames(classes.root, className)}>
       {points.map(({date, description}) => (
-        <Fragment key={description}>
+        <Fragment key={`${date}_${description}`}>
```
Comment (Will-Howard) [inline packages/lesswrong/components/ea-forum/giving-portal/Timeline.tsx:187]: Fix for duplicate key error

**Pull request merged**